src/test/java/org/esa/snap/dataio/RunProductReaderTest.java

```diff
@@ -0,0 +1,306 @@
+package org.esa.snap.dataio;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.esa.s2tbx.dataio.gdal.GDALLibraryInstaller;
+import org.esa.snap.core.dataio.ProductReader;
+import org.esa.snap.core.dataio.ProductReaderPlugIn;
+import org.esa.snap.core.datamodel.Product;
+import org.esa.snap.core.util.SystemUtils;
+import org.esa.snap.dataio.netcdf.NetCdfActivator;
+import org.esa.snap.jp2.reader.OpenJPEGLibraryInstaller;
+import org.esa.snap.runtime.LogUtils;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.rules.ErrorCollector;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static org.junit.Assert.fail;
+
+public class RunProductReaderTest {
```

Comment:
Why is this class necessary? Is ProductReaderAcceptanceTest not sufficient?

Reply:
The RunProductReaderTest is used to test certain product ids from the reader plugin JSON file. Also, the logger can be configured to enabled/disabled the messages for some classes to avoid unusefull messages in the console.

An example to run only the Muscate test for product id 'MUSCATE-v20':
-Dsnap.reader.tests.data.dir=\\CV-DEV-SRV01\Satellite_Imagery\TestingJUnitFiles
-Dsnap.reader.tests.class.name=org.esa.s2tbx.dataio.spot.SpotViewProductReaderPlugin
-Dsnap.reader.tests.product.ids=MUSCATE-v20
-Dorg.esa.snap.level=FINE
-Dorg.esa.snap.core.metadata.level=SEVERE
-Dorg.esa.snap.jp2.reader.internal.level=SEVERE
-Dsnap.reader.tests.failOnMissingData=false
-Drapid.eye.force.read.product.with.nitf.api=true
-Dorg.esa.s2tbx.dataio.rapideye.level=FINE

Reply:
Yes, to be able to specify the product IDs is a nice feature, but this would have been better adde to the ProductReaderAcceptanceTest class. and then documented on the [wiki page](https://senbox.atlassian.net/wiki/spaces/SNAP/pages/10878981/How+to+run+and+create+a+Reader+Test).

Same for the logging options.

Reply:
I decided to not modify the ProductReaderAcceptanceTest class and create a new class with implementation I need to run the tests for reading the products from the JSON file.
Of course, the implementation can be copied/moved in the ProductReaderAcceptanceTest class if it is needed there.

Reply:
Yes, it would be good if you can move into the ProductReaderAcceptanceTest.
I just want to avoid that we have several runners. That Luis creates next week another onot and the week after BC creates also one. And all implementations need to be maintained.
So better update and improve the existing one as long as it applicable. Of course very special needs can arise which make it necessary to create a new implementation. But then the name of the implementation should already expression the special case.